controllers/apps/v1beta4/update_pod_conditions.go

```diff
@@ -0,0 +1,132 @@
+			}
+		}
+
+		if _, ready := hash[corev1.ContainersReady]; ready {
+			if _, ok := clusterPods[pod.Name]; ok {
+				podOnServing := corev1.PodCondition{
+					Type:               appsv1beta4.PodOnServing,
+					Status:             corev1.ConditionTrue,
+					LastProbeTime:      metav1.Now(),
+					LastTransitionTime: metav1.Now(),
+				}
+				if pod.Labels["controller-revision-hash"] != instance.GetStatus().GetCurrentStatefulSetVersion() {
+					podOnServing.Status = corev1.ConditionFalse
+				}
+				if _, ok := instance.(*appsv1beta4.EmqxEnterprise); ok {
+					available, err := checkEmqxAvailability(&u, instance, &pod)
+					if err != nil {
+						return subResult{err: err}
+					}
+					if !available {
+						podOnServing.Status = corev1.ConditionFalse
+					}
+				}
+				exist := false
+				for k, condition := range pod.Status.Conditions {
+					if condition.Type == podOnServing.Type {
+						pod.Status.Conditions[k] = podOnServing
+						exist = true
+						break
+					}
+				}
+				if !exist {
+					pod.Status.Conditions = append(pod.Status.Conditions, podOnServing)
+				}
+
+				conditionsBytes, _ := json.Marshal(pod.Status.Conditions)
+				patchBytes := []byte(fmt.Sprintf(`{"status":{"conditions":%s}}`, string(conditionsBytes)))
+
+				err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+					return u.Client.Status().Patch(ctx, &pod, client.RawPatch(types.StrategicMergePatchType, patchBytes))
+				})
+				if err != nil {
+					return subResult{err: emperror.Wrap(err, "failed to update pod conditions")}
+				}
+			}
+		}
+	}
+	return subResult{}
+}
+
+func getPortForwardAPI(client client.Client, clientset *kubernetes.Clientset, config *rest.Config, instance appsv1beta4.Emqx, pod *corev1.Pod) (*portForwardAPI, error) {
+	o, err := innerPortFW.NewPortForwardOptions(clientset, config, pod, "8081")
+	if err != nil {
+		return nil, emperror.Wrap(err, "failed to create port forward options")
+	}
+
+	username, password, err := getBootstrapUser(context.Background(), client, instance)
+	if err != nil {
+		return nil, emperror.Wrap(err, "failed to get bootstrap user")
+	}
+	return &portForwardAPI{
+		Username: username,
+		Password: password,
+		Options:  o,
+	}, nil
+}
+
+func checkEmqxAvailability(u *updatePodConditions, instance appsv1beta4.Emqx, pod *corev1.Pod) (bool, error) {
+	p, err := getPortForwardAPI(u.Client, u.Clientset, u.Config, instance, pod)
+	if p == nil || err != nil {
+		return false, emperror.Wrap(err, "failed to get getPortForwardAPI")
+	}
+	defer close(p.Options.StopChannel)
+	if err := p.Options.ForwardPorts(); err != nil {
+		return false, emperror.Wrap(err, "failed to forward ports")
+	}
+
+	resp, _, err := p.requestAPI("GET", "api/v4/load_rebalance/availability_check", nil)
+	if err != nil {
+		return false, emperror.Wrap(err, "failed to check emqx availability")
+	}
+	if resp == nil || resp.StatusCode != 200 {
+		return false, emperror.Wrapf(err, "emqx pod %s-%s is unAvailable", pod.Namespace, pod.Name)
```

Comment:
If err is nil, emperror.Wrap returns nil.

Reply:
> If err is nil, emperror.Wrap returns nil.

i will replace Wrap with Errorf